### PR TITLE
Implement General sort

### LINKED ISSUE #48: General Sort
Add sorting feature for all fields, lexicographical order for fields with label data or numerical order for fields with quantitative data

### PR DESCRIPTION
**Pull request by laksherthanlife**
closes #48

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/SortCommand.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+/**
+ * Sorts all students in the address book based on any given field
+ */
+public class SortCommand extends Command {
+
+    public static final String COMMAND_WORD = "sort";
+    public static final String MESSAGE_SUCCESS = "Sorted all persons by %s in %s order.";
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": sorts all students in the address book "
+            + "based on any given field (case-insensitive) of the student in either ascending or descending order"
+            + "Parameters: <Field>(email, major, name, phone, star) <asc/desc> \n"
```
Comment (evedaktyl) [inline src/main/java/seedu/address/logic/commands/SortCommand.java:17]: Would it be meaningful to sort by bolt as well?

File: src/main/java/seedu/address/model/Model.java [hunk 2]
```diff
@@ -91,12 +91,12 @@ public interface Model {
      * @throws NullPointerException if {@code predicate} is null.
      */
     void updateFilteredStudentList(Predicate<Student> predicate);
```
Comment (evedaktyl) [inline src/main/java/seedu/address/model/Model.java:93]: Perhaps include access modifiers

File: src/main/java/seedu/address/logic/parser/SortCommandParser.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+        if (!VALID_FIELDS.contains(field)) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, SortCommand.MESSAGE_USAGE));
+        }
+
+        String direction = words[1].toLowerCase();
+        boolean isAscending = parseDirection(direction);
+        return new SortCommand(field, isAscending);
+    }
+
+    private boolean parseDirection(String direction) throws ParseException {
```
Comment (evedaktyl) [inline src/main/java/seedu/address/logic/parser/SortCommandParser.java:42]: Great SLAP.

File: src/test/java/seedu/address/logic/parser/SortCommandParserTest.java [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    }
+
+    @Test
+    void parse_spaces_throws() {
+        CommandParserTestUtil.assertParseFailure(PARSER, " ",
+                String.format(Messages.MESSAGE_INVALID_COMMAND_FORMAT, SortCommand.MESSAGE_USAGE));
+    }
+
+    @Test
+    void parse_onlyField_throws() {
```
Comment (evedaktyl) [inline src/test/java/seedu/address/logic/parser/SortCommandParserTest.java:24]: Would it be clearer to name this as parse_missingOrder_throws()?

**Review by evedaktyl — COMMENTED** [review]
LGTM, left some minor suggestions for valid fields and naming.

File: src/main/java/seedu/address/logic/commands/SortCommand.java (hunk 1 above)
Comment (c-wenlong) [reply src/main/java/seedu/address/logic/commands/SortCommand.java:17]: I think this one lets do it in 1.3b, other than that LGTM.

**Review by c-wenlong — COMMENTED** [review]
Good job @R-Laksh , ill merge it.

**Pull request merged**